misc/toml-parser/src/test/java/toml/parser/test/core/KeyValueTest.java

```diff
@@ -0,0 +1,130 @@
+/*
+ * Copyright (c) 2020, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
+ *
+ * WSO2 Inc. licenses this file to you under the Apache License,
+ * Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package toml.parser.test.core;
+
+import io.ballerina.toml.api.Toml;
+import io.ballerina.toml.ast.TomlArrayValueNode;
+import io.ballerina.toml.ast.TomlBooleanValueNode;
+import io.ballerina.toml.ast.TomlDoubleValueNodeNode;
+import io.ballerina.toml.ast.TomlLongValueNode;
+import io.ballerina.toml.ast.TomlStringValueNode;
+import org.testng.Assert;
+import org.testng.annotations.Test;
+
+import java.io.IOException;
+import java.io.InputStream;
+import java.util.ArrayList;
+import java.util.List;
+
+/**
+ * Basic Test for TOML Key Value Pairs.
+ */
+public class KeyValueTest {
+
+    @Test
+    public void testKeys() throws IOException {
+        InputStream inputStream = Thread.currentThread().getContextClassLoader()
+                .getResourceAsStream("core/keys.toml");
+        Toml read = Toml.read(inputStream);
+        String basicKey = ((TomlStringValueNode) read.get("key")).getValue();
+        String underscoreKey = ((TomlStringValueNode) read.get("underscore_key")).getValue();
+        String dashKey = ((TomlStringValueNode) read.get("dash-key")).getValue();
+        String intKey = ((TomlStringValueNode) read.get("1234")).getValue();
+
+        String dottedString =
+                ((TomlStringValueNode) read.getTable("127").getTable("0").getTable("0").get("1")).getValue();
+        String spacing = ((TomlStringValueNode) read.get("character encoding")).getValue();
+        String unicode = ((TomlStringValueNode) read.get("ʎǝʞ")).getValue();
+        String general = ((TomlStringValueNode) read.get("key2")).getValue();
+        String escape = ((TomlStringValueNode) read.get("quoted \"value\"")).getValue();
+
+        Assert.assertEquals(basicKey, "basic key");
+        Assert.assertEquals(underscoreKey, "Underscore Key");
+        Assert.assertEquals(dashKey, "Dash Key");
+        Assert.assertEquals(intKey, "Int Key");
+
+        Assert.assertEquals(dottedString, "value");
+        Assert.assertEquals(spacing, "value");
+        Assert.assertEquals(unicode, "value");
+        Assert.assertEquals(general, "value");
+        Assert.assertEquals(escape, "value");
```

Comment:
Hardcoding the asserts is not scaling in long term. Can we use the same approach as in the parser tests?

Reply:
Created. #26291